crates/paths/src/paths.rs

```diff
@@ -5,15 +5,45 @@ use std::sync::OnceLock;
 
 pub use util::paths::home_dir;
 
+/// A default editorconfig file name to use when resolving project settings.
+pub const EDITORCONFIG_NAME: &str = ".editorconfig";
+
+// Custom data directory if used.
+static CUSTOM_DATA_DIR: OnceLock<PathBuf> = OnceLock::new();
+
 /// Returns the relative path to the zed_server directory on the ssh host.
 pub fn remote_server_dir_relative() -> &'static Path {
     Path::new(".zed_server")
 }
 
+/// Sets a custom directory for all user data, overriding the default support directory.
+pub fn set_custom_data_dir(dir: &str) -> &'static PathBuf {
+    CUSTOM_DATA_DIR.get_or_init(|| {
+        let path = if cfg!(target_os = "windows") && (dir.contains(r"\") || dir.starts_with(r"\\")) {
+            // Windows absolute path (e.g., "C:\path" or "\\network\path")
+            PathBuf::from(dir)
+        } else if dir.starts_with("./") || dir.starts_with("../") || (!dir.starts_with("/") && !cfg!(target_os = "windows")) {
+            // Relative path on any OS (Linux/macOS: no leading "/", Windows: no drive letter or UNC)
+            std::env::current_dir()
```

Comment:
I don't think we need any of these branches, `std::create_dir_all` calls `libc::mkdir` in a loop, and that supports relative paths: https://man7.org/linux/man-pages/man2/mkdir.2.html

Reply:
updated